src/lib.rs

```diff
@@ -778,7 +783,7 @@ pub trait ConstantTimeLess: ConstantTimeEq + ConstantTimeGreater {
     /// ```
     #[inline]
     fn ct_lt(&self, other: &Self) -> Choice {
-        !self.ct_gt(other) & !self.ct_eq(other)
+        other.ct_gt(self)
```

Comment:
The current default definition assumes a total ordering over `Self`, which could introduce sneaky logic errors for users! It also makes two calls to comparison functions instead of one. I believe this new default definition will now be correct for partially-ordered `Self` and will only ever need to be modified for cases where `x < y` does not imply `y > x` (which I believe is a strict subset of partially ordered sets).